flake.nix

```diff
@@ -1,20 +1,56 @@
 {
+  description = "";
+
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    
     flake-utils.url = "github:numtide/flake-utils";
+
+    impermanence.url = "github:nix-community/impermanence";
+    
+    deploy-rs.url = "github:serokell/deploy-rs";
+    deploy-rs.inputs.nixpkgs.follows = "nixpkgs";
+    deploy-rs.inputs.utils.follows = "flake-utils";
   };
 
-  outputs = { nixpkgs, flake-utils, ... }:
-    flake-utils.lib.eachDefaultSystem (system: let
+  outputs = { self, nixpkgs, impermanence, deploy-rs, ... }: let
+    systems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+    # forEachSystem [ "x86_64-linux" ] (_: { example = true; }) -> { x86_64-linux.example = true }
+    forEachSystem = nixpkgs.lib.genAttrs systems;
```

Comment:
I believe this is the same as "forEachDefaultSystem"

Reply:
Wait no you removed the flake-utils reference entirely - could you drop the input and update the lockfile?

Reply:
New flake lock:
![flakelock](https://github.com/HacDC/nix-infra/assets/8916363/05851c57-fe67-4fe3-a7ed-275e52a672f3)